### ISSUE TITLE
Freeze encoder in ML Filter Classifier

### ISSUE THREAD

**Issue by florianmai** [Modalities/ml_filter#114]
While trying to reproduce the reported results from the FineWeb paper, I found that we reach consistently better results. Checking the paper again, I noticed that the FineWeb-Edu paper has a short note that they freeze the encoder when training, while we are finetuning the full model:

> To scale our filtering to the entirety of FineWeb, we trained a linear regression model on top of the
Snowflake-arctic-embed-m embedding model [65 ]. We fine-tuned this linear regressor on 410,000
of our Llama 3 synthetic annotations for 20 epochs with a learning rate of 3e-4 (**while keeping the
embedding and encoder layers frozen**). We selected the checkpoint with the highest F1 score on
the held-out validation set containing the remaining 50,000 samples, treating Llama 3 annotations
as ground-truth.

Did we miss this note or did we consciously decide not to freeze the encoder? If full finetuning reaches better results, is there any reason not to do it? Since the annotated data is sampled i.i.d. from the full dataset that we will annotate in ML3, it should generalize well.
